server/src/main/java/org/candlepin/model/ConsumerCurator.java

```diff
@@ -297,6 +298,28 @@ public Consumer lockAndLoad(Consumer c) {
 
     }
 
+    /**
+     * Find a consumer by uuid and immediately lock it.
+     *
+     * @param consumerUuid the uuid of the target consumer.
+     *
+     * @return the Consumer matching the given uuid, null if the consumer was not found.
+     */
+    @Transactional
```

Comment:
@mstead Perhaps the same principle applies as there though?
If the calling method is not have a @Transactional , this method might:

1. open the transaction
1. fetch and lock the consumer
1. close the transaction - and thereby lose the lock